src/backend/access/gin/gininsert.c

```diff
@@ -422,6 +422,7 @@ ginbuild(Relation heap, Relation index, IndexInfo *indexInfo)
 						  0, RelationGetNumberOfBlocks(index),
 						  true);
 		SetLastWrittenLSN(XactLastRecEnd, index->rd_smgr->smgr_rnode.node.relNode, 0, RelationGetNumberOfBlocks(index));
+		SetLastWrittenLSN(XactLastRecEnd, index->rd_smgr->smgr_rnode.node.relNode, REL_METADATA_PSEUDO_BLOCKNO, REL_METADATA_PSEUDO_BLOCKNO);
```

Comment:
This API where you have to call SetLastWrittenLSN twice, is a bit awkward. In all of these calls where you pass REL_METADATA_PSEUDO_BLOCKNO, you have just loaded the relation with data. How replacing SetLastWrittenLSN() with two functions:
```
SetLastWrittenLSNForRelation(Relation, ForkNumber, XLogRecPtr);
SetLastWrittenLSNForBlock(RelFileNode, ForkNumber, BlockNumber, XLogRecPtr);
```